app/docs/[...slug]/page.tsx

```
@@ -0,0 +1,227 @@
+// Cache the flattened docs structure
+const getFlatDocs = cache(() => {
+  function flattenDocs(
+    structure: {
+      title: string;
+      path: string;
+      children?: { title: string; path: string }[];
+    }[]
+  ): { title: string; path: string }[] {
+    return structure.reduce<{ title: string; path: string }[]>((acc, item) => {
+      if (item.children?.length) {
+        return [...acc, ...item.children];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)